diff --git a/modules/facets_range_widget/README.txt b/modules/facets_range_widget/README.txt
index e4a9e49..448bba9 100644
--- a/modules/facets_range_widget/README.txt
+++ b/modules/facets_range_widget/README.txt
@@ -6,20 +6,10 @@ CONTENTS OF THIS FILE
 Installation
 ------------
 
- * If you want to use the sliders, you need to add the Slider pips jquery
-   plugin:
-    - create the /libraries/jquery-ui-slider-pips/dist folder.
-    - download the following files from
-    https://github.com/simeydotme/jQuery-ui-Slider-Pips/tree/v1.11.3/dist
-       - jquery-ui-slider-pips.min.js
-       - jquery-ui-slider-pips.min.css
-
-   You can find more information about this jquery plugin on
-   http://simeydotme.github.io/jQuery-ui-Slider-Pips/
-
-   Alternatively, you can install from Asset Packagist using Composer.
+ * If you want to use the sliders, you need to add the noUiSlider library
+   from Asset Packagist using Composer.
    If you are using the Lightning or Drupal Commerce base distros, just run
-   `composer require "bower-asset/jquery-ui-slider-pips:^1.11"
+   `composer require "npm-asset/nouislider:^14"`
    If you don't have Asset Packagist configured, see
    https://github.com/drupal-composer/drupal-project/issues/278#issuecomment-300714410
    for instructions.
diff --git a/modules/facets_range_widget/facets_range_widget.libraries.yml b/modules/facets_range_widget/facets_range_widget.libraries.yml
index b647f24..5996c44 100644
--- a/modules/facets_range_widget/facets_range_widget.libraries.yml
+++ b/modules/facets_range_widget/facets_range_widget.libraries.yml
@@ -1,17 +1,17 @@
-jquery.ui.slider.pips:
-  remote: http://simeydotme.github.io/jQuery-ui-Slider-Pips/
-  version: v1.11.3
+nouislider:
+  remote: https://refreshless.com/nouislider/
+  version: 15.1.1
   license:
     name: MIT
-    url: https://github.com/simeydotme/jQuery-ui-Slider-Pips/blob/v1.11.3/README.md
+    url: https://github.com/leongersen/noUiSlider/blob/master/LICENSE.md
     gpl-compatible: true
   js:
-    /libraries/jquery-ui-slider-pips/dist/jquery-ui-slider-pips.min.js: { minified: true, attributes: { defer: true } }
+    /libraries/nouislider/distribute/nouislider.min.js: { minified: true }
   css:
     component:
-      /libraries/jquery-ui-slider-pips/dist/jquery-ui-slider-pips.min.css: { minified: true }
+      /libraries/nouislider/distribute/nouislider.min.css: { minified: true }
   dependencies:
-    - core/jquery.ui.slider
+    - facets_range_widget/wnumb
 
 slider:
   version: VERSION
@@ -23,6 +23,4 @@ slider:
   dependencies:
     - core/drupal
     - core/drupalSettings
-    - core/jquery.once
-    - facets_range_widget/jquery.ui.slider.pips
-    - core/jquery.ui.touch-punch
+    - facets_range_widget/nouislider
diff --git a/modules/facets_range_widget/js/slider.js b/modules/facets_range_widget/js/slider.js
index 916e71f..659526f 100644
--- a/modules/facets_range_widget/js/slider.js
+++ b/modules/facets_range_widget/js/slider.js
@@ -3,8 +3,7 @@
  * Provides the slider functionality.
  */
 
-(function ($) {
-
+ (function () {
   'use strict';
 
   Drupal.facets = Drupal.facets || {};
@@ -12,37 +11,48 @@
   Drupal.behaviors.facet_slider = {
     attach: function (context, settings) {
       if (settings.facets !== 'undefined' && settings.facets.sliders !== 'undefined') {
-        $.each(settings.facets.sliders, function (facet, settings) {
-          Drupal.facets.addSlider(facet, settings);
-        });
+        for (const [facet, facet_settings] of Object.entries(settings.facets.sliders)) {
+          Drupal.facets.addSlider(document.getElementById(facet), facet_settings);
+        }
       }
     }
   };
 
   Drupal.facets.addSlider = function (facet, settings) {
-    var defaults = {
-      stop: function (event, ui) {
+    noUiSlider.create(facet, {
+      range:{
+        min: 0,
+        max: settings.labels.length - 1
+      },
+      step: 1,
+      start:[
+        settings.values[0],
+        settings.values[1]
+      ],
+      connect: true,
+      tooltips: true,
+      format: {
+        to: function (value) {
+          // Converts handle values into tooltips.
+          return settings.prefix + settings.labels[Math.round(value)] + settings.suffix;
+        },
+        from: function (value) {
+          // Converts the starting values into slider values for initial handle positions.
+          return settings.labels.indexOf(value);
+        }
+      }
+    });
+    facet.noUiSlider.on('change', function(values, handle, unencoded, tap, positions, noUiSlider){
         if (settings.range) {
-          window.location.href = settings.url.replace('__range_slider_min__', ui.values[0]).replace('__range_slider_max__', ui.values[1]);
+          window.location.href = settings.url.replace('__range_slider_min__', settings.labels[Math.round(unencoded[0])]).replace('__range_slider_max__', settings.labels[Math.round(unencoded[1])]);
         }
         else {
-          window.location.href = settings.urls['f_' + ui.value];
+          window.location.href = settings.urls['f_' + values[0]];
         }
-      }
-    };
-
-    $.extend(defaults, settings);
-
-    $('[id^="' + facet +'"][id$="' + facet +'"]').slider(defaults)
-    .slider('pips', {
-      prefix: settings.prefix,
-      suffix: settings.suffix
-    })
-    .slider('float', {
-      prefix: settings.prefix,
-      suffix: settings.suffix,
-      labels: settings.labels
     });
-  };
 
-})(jQuery);
+    // Make the noUiSlider object availble for sites to add their own handlers.
+    window.facet_nouisliders = window.facet_nouisliders || {};
+    window['facet_nouisliders'][facet.getAttribute('id')] = facet;
+  };
+})();
\ No newline at end of file
